streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimestampedStoreTest.java

```diff
@@ -0,0 +1,619 @@
+        entries.add(new KeyValue<>("1".getBytes(UTF_8), null));
+
+        context.restore(rocksDBStore.name(), entries);
+
+        final KeyValueIterator<Bytes, byte[]> iterator = rocksDBStore.all();
+        final Set<String> keys = new HashSet<>();
+
+        while (iterator.hasNext()) {
+            keys.add(stringDeserializer.deserialize(null, iterator.next().key.get()));
+        }
+
+        assertThat(keys, equalTo(Utils.mkSet("2", "3")));
+    }
+
+    @Test
+    public void shouldThrowNullPointerExceptionOnNullPut() {
+        rocksDBStore.init(context, rocksDBStore);
+        try {
+            rocksDBStore.put(null, stringSerializer.serialize(null, "someVal"));
+            fail("Should have thrown NullPointerException on null put()");
+        } catch (final NullPointerException e) {
+            // this is good
+        }
+    }
+
+    @Test
+    public void shouldThrowNullPointerExceptionOnNullPutAll() {
+        rocksDBStore.init(context, rocksDBStore);
+        try {
+            rocksDBStore.put(null, stringSerializer.serialize(null, "someVal"));
+            fail("Should have thrown NullPointerException on null put()");
+        } catch (final NullPointerException e) {
+            // this is good
+        }
+    }
+
+    @Test
+    public void shouldThrowNullPointerExceptionOnNullGet() {
+        rocksDBStore.init(context, rocksDBStore);
+        try {
+            rocksDBStore.get(null);
+            fail("Should have thrown NullPointerException on null get()");
+        } catch (final NullPointerException e) {
+            // this is good
+        }
+    }
+
+    @Test
+    public void shouldThrowNullPointerExceptionOnDelete() {
+        rocksDBStore.init(context, rocksDBStore);
+        try {
+            rocksDBStore.delete(null);
+            fail("Should have thrown NullPointerException on deleting null key");
+        } catch (final NullPointerException e) {
+            // this is good
+        }
+    }
+
+    @Test
+    public void shouldThrowNullPointerExceptionOnRange() {
+        rocksDBStore.init(context, rocksDBStore);
+        try {
+            rocksDBStore.range(null, new Bytes(stringSerializer.serialize(null, "2")));
+            fail("Should have thrown NullPointerException on deleting null key");
+        } catch (final NullPointerException e) {
+            // this is good
+        }
+    }
+
+    @Test(expected = ProcessorStateException.class)
+    public void shouldThrowProcessorStateExceptionOnPutDeletedDir() throws IOException {
+        rocksDBStore.init(context, rocksDBStore);
+        Utils.delete(dir);
+        rocksDBStore.put(
+            new Bytes(stringSerializer.serialize(null, "anyKey")),
+            stringSerializer.serialize(null, "anyValue"));
+        rocksDBStore.flush();
+    }
+
+    @Test
+    public void shouldMigrateDataFromDefaultToTimestampColumnFamily() throws Exception {
+        prepareOldStore();
```

Comment:
This test let checkstyle fail because method was too long. Extracted some part into `prepareOldStore()` to fix checkstyle.